src/OSPSuite.SimModelNative/src/DESolver.cpp

```diff
@@ -674,27 +686,31 @@ namespace SimModelNative
 	Sensitivity_Rhs_Return_Value DESolver::ODESensitivityRhsFunction(double t, const double * y, double * ydot,
 		int iS, const double * yS, double * ySdot, void * f_data)
 	{
-		throw "Not implemented yet";
-
-//		//fill the RHS of the ODE variables first
-//		Rhs_Return_Value odeRHSReturnValue = ODERhsFunction(t, y, ydot, f_data);
-//
-//		if (odeRHSReturnValue == RHS_FAILED)
-//			return SENSITIVITY_RHS_FAILED;
-//
-//		if (odeRHSReturnValue == RHS_RECOVERABLE_ERROR)
-//			return SENSITIVITY_RHS_RECOVERABLE_ERROR;
-//
-//		//just in case (return value should be ok now)
-//		if (odeRHSReturnValue != RHS_OK)
-//			throw ErrorData(ErrorData::ED_ERROR, ERROR_SOURCE, "ODERhsFunction return unknown exit code");
-//
+		//fill the RHS of the ODE variables first
+		Rhs_Return_Value odeRHSReturnValue = ODERhsFunction(t, y, yS, ydot, f_data);
+
+		if (odeRHSReturnValue == RHS_FAILED)
+			return SENSITIVITY_RHS_FAILED;
+
+		if (odeRHSReturnValue == RHS_RECOVERABLE_ERROR)
+			return SENSITIVITY_RHS_RECOVERABLE_ERROR;
+
+		//just in case (return value should be ok now)
+		if (odeRHSReturnValue != RHS_OK)
+			throw ErrorData(ErrorData::ED_ERROR, "ODESensitivityRhsFunction", "ODERhsFunction return unknown exit code");
+
 //		throw ErrorData(ErrorData::ED_ERROR, ERROR_SOURCE, "ODESensitivityRhsFunction not implemented yet");
-//
-//		//-------------------------------------------------
-//		// 1. Get cached values
-//		//-------------------------------------------------
 
+		//-------------------------------------------------
+		// 1. Get cached values
+		//-------------------------------------------------
+		for (int iEquation = 0; iEquation < m_ODE_NumUnknowns; iEquation++)
+		{
+			ySdot[iEquation] = 
+				m_ODEVariables[iEquation]->JacobianStateVariableFor(iEquation)->DE_Compute(y, t, ScaleFactorUsageMode::USE_SCALEFACTOR) * yS[iEquation] +
+				m_ODEVariables[iEquation]->JacobianParameterFor(_sensitivityParameters[iS]->GetId())->DE_Compute(y, t, ScaleFactorUsageMode::USE_SCALEFACTOR);
```

Comment:
That's wrong, it must be something like this:
```
ySdot[iEquation] = 0;
for(odeIdx=0; odeIdx < m_ODE_NumUnknowns; odeIdx++)
{
   ySdot[iEquation] += m_ODEVariables[iEquation]->JacobianStateVariableFor(odeIdx)->DE_Compute(y, t, ScaleFactorUsageMode::USE_SCALEFACTOR) * yS[odeIdx];
}
ySdot[iEquation] += m_ODEVariables[iEquation]->JacobianParameterFor(_sensitivityParameters[iS]->GetId())->DE_Compute(y, t, ScaleFactorUsageMode::USE_SCALEFACTOR);
```

E.g. compare the calculation for the [2.2 A serial dense example: cvsRoberts FSA dns](https://raw.githubusercontent.com/LLNL/sundials/master/doc/cvodes/cvs_examples.pdf):

[cvsRoberts_FSA_dns.zip](https://github.com/Open-Systems-Pharmacology/OSPSuite.SimModel/files/8472577/cvsRoberts_FSA_dns.zip)

<details><summary><b>Expand<b></summary>

![grafik](https://user-images.githubusercontent.com/25061876/162956537-62f07196-89a7-45dd-b370-700bd262a08c.png)

```
static int fS(int Ns, realtype t, N_Vector y, N_Vector ydot,
              int iS, N_Vector yS, N_Vector ySdot,
              void *user_data, N_Vector tmp1, N_Vector tmp2)
{
  UserData data;
  realtype p1, p2, p3;
  realtype y1, y2, y3;
  realtype s1, s2, s3;
  realtype sd1, sd2, sd3;

  data = (UserData) user_data;
  p1 = data->p[0]; p2 = data->p[1]; p3 = data->p[2];

  y1 = Ith(y,1);  y2 = Ith(y,2);  y3 = Ith(y,3);
  s1 = Ith(yS,1); s2 = Ith(yS,2); s3 = Ith(yS,3);

  sd1 = -p1*s1 + p2*y3*s2 + p2*y2*s3;
  sd3 = 2*p3*y2*s2;
  sd2 = -sd1-sd3;

  switch (iS) {
  case 0:
    sd1 += -y1;
    sd2 +=  y1;
    break;
  case 1:
    sd1 +=  y2*y3;
    sd2 += -y2*y3;
    break;
  case 2:
    sd2 += -y2*y2;
    sd3 +=  y2*y2;
    break;
  }

  Ith(ySdot,1) = sd1;
  Ith(ySdot,2) = sd2;
  Ith(ySdot,3) = sd3;

  return(0);
}
```
</details>

Reply:
thx... I have been struggling with it